test/conformance/source/environment.cpp

```diff
@@ -199,6 +201,15 @@ DevicesEnvironment::DevicesEnvironment(int argc, char **argv)
         error = "Could not find any devices associated with the platform";
         return;
     }
+    // Get the argument (NUMBER_OF_DEVICES) to run tests on the wanted number of devices
+    int count_set = 0;
+    count_set = std::atoi(argv[2]);
+    if (count_set < 0 || count_set > std::numeric_limits<uint32_t>::max()) {
+        error = "Invalid NUMBER_OF_DEVICES argument";
+        return;
+    } else if (count_set > 0) {
```

Comment:
You don't handle a special case of `0` here. As you described, all available adapters should be considered.

Reply:
I believe it's on purpose (`0` makes no limitation on dev count)... but I agree, that you should at least document it here

Reply:
The `devices` vector is resized to zero in this case and then it's data is passed to the `urDeviceGet` function. That's possibly a nullptr, so none of the devices handles will be added to this vec.

Reply:
no, it's not 😉 it's the `count_set` that's equal to `0`, not `count` itself

Reply:
You're right :) It works well then.